### DISCUSSION TITLE
unbale to use math functions in pnp search results webpart custom template

### DISCUSSION THREAD

**Discussion by GarimaTe** [microsoft-search/pnp-modern-search#2827] [Q&A] [answered]
I am trying to create this custom template which places 2 cards in a row but the math function is not working. Any help is appreciated

My code is as below:

<content id="data-content">
  <style>
    /* Insert your CSS overrides here */
    .example-themePrimary a {
      color: {
         {
          @root.theme.palette.themeprimary
        }
      }
    }

    .myfirstcss{
        font-size: 30px;
        font-weight: 600;
    }
section {
    white-space: nowrap;
}
.ms-DocumentCard {
  border: 1px solid #ccc;
  box-shadow: 0 0 8px rgba(0, 0, 0, 0.15);
  width: 150px;
  height: 250px;
  margin: 0 5px 20px 5px;
  display: inline-block;
 
}
.ms-DocumentCard-details{
padding: 20px;
text-align: left;
}

.ms-DocumentCardLocation-link{
 font-weight: bold;
 font-size: 14px;
  font-family: Verdana, sans-serif;
}

.card-inline {
   margin-bottom: 10px; 
}
.choice-pill {
  display: inline-block;
  padding: 4px 8px;
  border-radius: 12px;
  color: white;
  background-color: #89CFF0; 
}

   
  </style>
  

  <div class="template">
    {{#if @root.properties.showSelectedFilters}}
    <pnp-selectedfilters
      data-filters="{{JSONstringify filters.selectedFilters 2}}"
      data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}"
      data-instance-id="{{filters.instanceId}}"
      data-operator="{{filters.filterOperator}}"
      data-theme-variant="{{JSONstringify @root.theme}}"
    ></pnp-selectedfilters>
    {{/if}} {{#if @root.properties.showResultsCount}}
    <div class="template--resultCount">
      <label class="ms-fontWeight-semibold"
        >{{getCountMessage @root.data.totalItemsCount
        @root.inputQueryText}}</label>
    </div>
    {{/if}}

   {{!-- <div class="template--cardContainer" > --}}

      {{#each data.items as |item|}} {{#> resultTypes item=item}} 
	{{#if (eq (mathMod index 2) 0)}} 
      <div class="row">
	   
    {{/if}} 
         <div class="col-md-6"> 
		       
  
   <div class='ms-DocumentCard' >
		 <div class='ms-DocumentCard-details'>
		 
		 <pnp-iconfile class="icon"
          data-extension="{{slot item @root.slots.FileType}}"
         data-theme-variant="{{JSONstringify @root.theme}}"
		 data-size="32">
		  </pnp-iconfile>
		  
		 <div class="ms-DocumentCardLocation card-inline">
		  <span class="ms-DocumentCardLocation-icon ms-Icon ms-Icon--OpenFile card-inline"></span>
      <a class="ms-DocumentCardLocation-link" href="{{slot item @root.slots.Path}}?web=1" target="_blank">{{slot item @root.slots.Title}}</a></div>
		 
		<div class="ms-DocumentCard-description card-inline choice-pill">{{slot item @root.slots.Functions}}</div>
		
		 <div class="ms-DocumentCard-author card-inline ">{{itemAt (split (slot item @root.slots.Author) "| ") 1}}</div>
		
		<div class="ms-DocumentCard-status card-inline">{{getDate (slot item @root.slots.Date) 'LL'}}</div>
		
     
    
		 </div>
	 </div>  
   </div>
	{{#if (or (eq (mathMod index 2) 1) (eq index (sub (length data.items) 1))))}} 
           </div> 
	{{/if}} 
  
      {{/resultTypes}} {{/each}}
    </div>
	</div>
  
</content>``

**Comment by kasperbolarsen** [comment]
are you looking for an Even/ Odd function like this : https://stackoverflow.com/questions/18976274/odd-and-even-number-comparison-helper-for-handlebars ?

**Reply by calebmil** [reply]
@kasperbolarsen is this type of helper built in to Modern Search now or do we have to write an extension?  Seems like one of those things that would be useful as part of the core solution.

**Reply by kasperbolarsen** [reply, marked as answer]
Even / Odd is currently a custom layout as I haven't yet figured out how to add it to the project,  as it will require that we add a color picker to the web part 

but feel free to reuse as much as possible:
https://github.com/microsoft-search/pnp-modern-search-layouts/tree/main/Results/Handlebars/List%20Alternating%20background%20color


sorry, I had to read the question again, and this time I got it: We already have a MOD helper in Modern Search, 
                    {{#eq (modulo @index 2) 0  }}

**Comment by calebmil** [comment]
Thanks @kasperbolarsen!  My customer wanted a way to stack state abbreviations from a multi-value column to make the layout easier to read.  

The field was rendering like this when rendered directly from the managed property:
AK;AR;AZ;CA;CT...

**Stacked Values**
The code below transformed this to a stack of values like this:
AK
AR
AZ
CA
CT
DC
DE
```
<div class="template--listItem--tags example-themePrimary">
   {{#if (slot item @root.slots.StateCodes)}}
       <div>
           {{#each (split (slot item @root.slots.StateCodes) ";") as |state| }}
               <div class="Simpletags">
                   {{trim state}}
               </div>
           {{/each}}
       </div>
   {{/if}}
</div>
```
**Stacked with multiple values per line**
Finally, the following code took the stack of values and rendered them three-per line to take up less space vertically like this:
AK AR AZ
CA CT DC
DE ...

```
<div class="template--listItem--tags example-themePrimary">
   {{#if (slot item @root.slots.StateCode)}}
       <div>
           {{#each (split (slot item @root.slots.StateCodes) ";") as |state| }}
              
                   {{#eq (modulo @index 3) 0 }} 
                       <div class="Simpletags">
                   {{/eq}}
                       <span role="gridcell" style="align:center; width:32%;" >
                           {{trim state}}
                       </span>
                   {{#eq (modulo @index 3) 2 }}
                       </div>
                   {{/eq}}
               
           {{/each}}
       </div>
   {{/if}}
</div>
```
